packages/twenty-front/src/modules/object-record/object-options-dropdown/components/ObjectOptionsDropdownContent.tsx

```diff
@@ -0,0 +1,184 @@
+    recordBoardId: recordIndexId,
+    viewBarId: recordIndexId,
+  });
+
+  const { viewGroupFieldMetadataItem } = useRecordGroups({
+    objectNameSingular: objectMetadataItem.nameSingular,
+  });
+
+  const { openObjectRecordsSpreasheetImportDialog } =
+    useOpenObjectRecordsSpreadsheetImportDialog(
+      objectMetadataItem.nameSingular,
+    );
+
+  const { progress, download } = useExportRecords({
+    delayMs: 100,
+    filename: `${objectMetadataItem.nameSingular}.csv`,
+    objectMetadataItem,
+    recordIndexId,
+    viewType,
+  });
+
+  return (
+    <>
+      {!currentContentId && (
+        <>
+          <DropdownMenuHeader StartIcon={IconList}>
+            {objectMetadataItem.labelPlural}
+          </DropdownMenuHeader>
+          {/** TODO: Should be removed when view settings contains more options */}
+          {viewType === ViewType.Kanban && (
+            <>
+              <DropdownMenuItemsContainer>
+                <MenuItem
+                  onClick={() => onContentChange('viewSettings')}
+                  LeftIcon={IconLayout}
+                  text="View settings"
+                  hasSubMenu
+                />
+              </DropdownMenuItemsContainer>
+              <DropdownMenuSeparator />
+            </>
+          )}
+          <DropdownMenuItemsContainer>
+            <MenuItem
+              onClick={() => onContentChange('fields')}
+              LeftIcon={IconTag}
+              text="Fields"
+              contextualText={`${visibleBoardFields.length} shown`}
+              hasSubMenu
+            />
+            <MenuItem
+              onClick={() => onContentChange('recordGroups')}
+              LeftIcon={IconLayoutList}
+              text="Group by"
+              contextualText={viewGroupFieldMetadataItem?.label}
+              hasSubMenu
+            />
+          </DropdownMenuItemsContainer>
+          <DropdownMenuSeparator />
+          <DropdownMenuItemsContainer>
+            <MenuItem
+              onClick={download}
+              LeftIcon={IconFileExport}
+              text={displayedExportProgress(progress)}
+            />
+            <MenuItem
+              onClick={() => openObjectRecordsSpreasheetImportDialog()}
+              LeftIcon={IconFileImport}
+              text="Import"
+            />
+            <MenuItem
+              onClick={() => {
+                handleToggleTrashColumnFilter();
+                toggleSoftDeleteFilterState(true);
+                closeDropdown();
+              }}
+              LeftIcon={IconRotate2}
+              text={`Deleted ${objectNamePlural}`}
+            />
+          </DropdownMenuItemsContainer>
+        </>
+      )}
+
+      <ObjectOptionsDropdownViewSettingsContent
+        viewType={viewType}
+        recordIndexId={recordIndexId}
+        objectMetadataItem={objectMetadataItem}
+      />
+
+      <ObjectOptionsDropdownFieldsContent
+        viewType={viewType}
```

Comment:
we are drilling down viewType, recordIndexId, objectMetadataItem, we could introduce an ObjectOptionsDrodownContext if this feels useful. I'm fine with the current implementation too!